### PR TITLE
Animations Component the number :100: 

### PR DESCRIPTION
**Pull request by Octopus773**
## Proposed changes

The animation component allows you to play animations from a file and tha anim index

## Information
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Non-breaking changes

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,9 +80,7 @@ add_executable(unit_tests EXCLUDE_FROM_ALL
 	tests/MainTest.cpp
 	tests/EngineTests.cpp
 	tests/CallbackTest.cpp
-	tests/CollisionTest.cpp
```
Comment (AnonymusRaccoon) [inline CMakeLists.txt:78]: Why do you remove tests?

File: lib/Ray/sources/Model/ModelAnimations.cpp [hunk 2]
```diff
@@ -10,21 +10,32 @@
...
 RAY::ModelAnimations::ModelAnimations(const std::string &filePath):
-	_animationsPtr(_animationsCache.fetch(filePath, &this->_animationCount))
+	_animationsPtr(_animationsCache.fetch(filePath, &this->_animationCount)),
+	_filePath(filePath)
 {
 	::ModelAnimation *ptr = this->_animationsPtr.get();
 
 	for (int i = 0; i < this->_animationCount; i++)
-		this->_animations.push_back(RAY::ModelAnimation(ptr[i]));
+		this->_animations.emplace_back(RAY::ModelAnimation(ptr[i]));
```
Comment (AnonymusRaccoon) [inline lib/Ray/sources/Model/ModelAnimations.cpp:19]: emplace_back(ptr[i]), here you call a copy constructor.

File: lib/Ray/sources/Model/ModelAnimations.cpp [hunk 3]
```diff
@@ -10,21 +10,32 @@
...
 }
 
 RAY::ModelAnimation &RAY::ModelAnimations::operator[](int index)
 {
-	return this->_animations[index];
+	return this->_animations.at(index);
```
Comment (AnonymusRaccoon) [inline lib/Ray/sources/Model/ModelAnimations.cpp:24]: Why?

File: lib/Ray/sources/Model/ModelAnimations.hpp [hunk 4]
```diff
@@ -27,15 +27,24 @@ namespace RAY {
...
+			//! @brief Default move ctor
+			ModelAnimations(ModelAnimations &&) = default;
+
 			//! @brief Only single entity can hold these animations pointers
 			ModelAnimations &operator=(const ModelAnimations &) = delete;
 
 			//! @brief Castin Object to raw model animation pointer
 			ModelAnimation &operator[](int index);
 
+			//! @brief Same usage as the operator[] but const
```
Comment (AnonymusRaccoon) [inline lib/Ray/sources/Model/ModelAnimations.hpp:39]: You can do a operator[] const for this purpose.

File: lib/Ray/sources/Utils/Cache.hpp [hunk 5]
```diff
@@ -60,11 +60,13 @@ namespace RAY {
 			{};
 			std::shared_ptr<::ModelAnimation> fetch(const std::string &path, int *counter)
 			{
+				::ModelAnimation *animations = this->_dataLoader(path.c_str(), counter);
```
Comment (AnonymusRaccoon) [inline lib/Ray/sources/Utils/Cache.hpp:63]: That's a big no, you load the resource even if the resource is already in the cache.

File: sources/Component/Animation/AnimationsComponent.cpp [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+		  _currentAnimIndex(animIndex),
+		  _animDisabled(play)
+	{
+		this->_modelAnimation[this->_currentAnimIndex].setFrameCounter(0);
+	}
+
+	WAL::Component *AnimationsComponent::clone(WAL::Entity &entity) const
+	{
+		return new AnimationsComponent(entity,
+								 RAY::ModelAnimations(this->_modelAnimation.getFilePath()),
```
Comment (AnonymusRaccoon) [inline sources/Component/Animation/AnimationsComponent.cpp:23]: Nice indentation x)

File: sources/System/Animation/AnimationsSystem.cpp [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+#include "Component/Animation/AnimationsComponent.hpp"
+#include "Model/Model.hpp"
+#include "Component/Renderer/Drawable3DComponent.hpp"
+
+namespace BBM
+{
+
+	AnimationsSystem::AnimationsSystem()
+		: WAL::System({
+		    typeid(Drawable3DComponent<RAY::Drawables::Drawables3D::Model>),
```
Comment (AnonymusRaccoon) [inline sources/System/Animation/AnimationsSystem.cpp:16]: Indent

**Review by AnonymusRaccoon — COMMENTED** [review]
Actually, the animation is not changed based on what the player does, is it?

File: CMakeLists.txt (hunk 1 above)
Comment (Octopus773) [reply CMakeLists.txt:78]: merge issue fixed

File: lib/Ray/sources/Model/ModelAnimations.cpp (hunk 3 above)
Comment (Octopus773) [reply lib/Ray/sources/Model/ModelAnimations.cpp:24]: at is more explicit

File: lib/Ray/sources/Model/ModelAnimations.hpp (hunk 4 above)
Comment (Octopus773) [reply lib/Ray/sources/Model/ModelAnimations.hpp:39]: nope operator[] is  should not be const

File: lib/Ray/sources/Utils/Cache.hpp (hunk 5 above)
Comment (Octopus773) [reply lib/Ray/sources/Utils/Cache.hpp:63]: didn't saw that was changed during a valgrind fix
fixed

File: sources/Component/Animation/AnimationsComponent.cpp (hunk 6 above)
Comment (Octopus773) [reply sources/Component/Animation/AnimationsComponent.cpp:23]: xD
my view
image

File: sources/System/Animation/AnimationsSystem.cpp (hunk 7 above)
Comment (Octopus773) [reply sources/System/Animation/AnimationsSystem.cpp:16]: fixed

**Review by AnonymusRaccoon — APPROVED** [review]

**Review by TrueBabyChaise — APPROVED** [review]

**Review by Arthi-chaud — APPROVED** [review]

**Review by EternalRat — APPROVED** [review]

**Pull request merged**